src/Assets/Javascript/locations-map.js

```diff
@@ -1,17 +1,116 @@
+const createPopupClass = () => {
+  const panToWithOffset = function(map, latlng, offsetX, offsetY) {
+    const ov = new google.maps.OverlayView();
+    ov.onAdd = function() {
+      const proj = this.getProjection();
+      const aPoint = proj.fromLatLngToContainerPixel(latlng);
+      aPoint.x = aPoint.x + offsetX;
+      aPoint.y = aPoint.y + offsetY;
+      map.panTo(proj.fromContainerPixelToLatLng(aPoint));
+    };
+    ov.draw = function() {};
+    ov.setMap(map);
+  };
+
+  // Based on: https://developers.google.com/maps/documentation/javascript/examples/overlay-popup
+  /**
+   * A customized popup on the map.
+   * @param {!google.maps.LatLng} position
+   * @param {!Element} content
+   * @constructor
+   * @extends {google.maps.OverlayView}
+   */
+  const Popup = function(position, closedContent, openContent) {
+    this.position = position;
+    const content = document.createElement("div");
+    content.classList.add("popup-bubble-content");
```

Comment:
The CSS classes are a bit of a mess for this component... I wrote them hastily. We should refactor them into proper BEM so they respect the `block > element` hierarchy.